.github/workflows/release.yaml

```
@@ -0,0 +1,141 @@
+---
+name: "Create the release"
+
+on:
+  release:
+    types: ["created", "published", "prereleased"]
+  workflow_dispatch:
+
+jobs:
+  gather-informations:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.get_tags.outputs.VERSION }}
+      release: ${{ steps.get_tags.outputs.RELEASE }}
+      tag: ${{ steps.get_tags.outputs.TAG }}
+    steps:
+      - name: Get Latest Release and List Artifacts
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        id: get_latest_release
+        uses: actions/github-script@v6
+        with:
+          script: |
+            // Retrieve the latest release
+            const latestRelease = await github.rest.repos.getLatestRelease({
+              owner: context.repo.owner,
+              repo: context.repo.repo
+            });
+
+            // Retrieve informations from the latest release
+            const assets = latestRelease.data.assets.map(asset => asset.name);
+            let version = latestRelease.data.tag_name.replace(/^v/i, '');
+            let release = 1
+            if (assets.length === 0) {
+              console.log("No artifacts attached to the latest release. Using tag name as version.");
+            } else {
+              const firstAsset = assets[0]; // Retrieve first element of the list
+              const regex = /([0-9.]{3,})-([0-9]+)/;
+              const match = firstAsset.match(regex);
+              release = match ? match[2] : ''; // Retrieve the last release version using artifact name
+              if (release !== '') {
+                release = (parseInt(release) + 1).toString();
+              } else {
+                release = 1
+              }
+              version = match ? match[1] : '';
+            }
+
+            // Set as output all necessary values
+            core.setOutput("release", release);
+            core.setOutput("version", version);
+            core.setOutput("tag", latestRelease.data.tag_name);
+
+      - name: Get version and release Tag
+        id: get_tags
+        run: |
+          if [ "${{ github.event_name }}" == "workflow_dispatch" ]; then
+            if [ -z ${{ steps.get_latest_release.outputs.version }} ]; then 
+              RAW_VERSION=${{ steps.get_latest_release.outputs.tag }}
+              CLEAN_VERSION=${RAW_VERSION#v}
+            else 
+              CLEAN_VERSION=${{ steps.get_latest_release.outputs.version }}
+            fi
+            echo "VERSION=$CLEAN_VERSION" >> $GITHUB_OUTPUT
+            echo "RELEASE=${{ steps.get_latest_release.outputs.release }}" >> $GITHUB_OUTPUT
+            echo "TAG=${{ steps.get_latest_release.outputs.tag }}" >> $GITHUB_OUTPUT
+          else
+            RAW_VERSION=${{ github.event.release.tag_name }}
+            CLEAN_VERSION=${RAW_VERSION#v}
+            echo "VERSION=$CLEAN_VERSION" >> $GITHUB_OUTPUT
+            echo "RELEASE=1" >> $GITHUB_OUTPUT
+            echo "TAG=$RAW_VERSION" >> $GITHUB_OUTPUT
+          fi
```

Proposed some refactor:

```suggestion
    - name: Get release, tag and version
      id: get_release_tag_version
      run: |
        if [ "${{ github.event_name }}" == "workflow_dispatch" ]; then
          RELEASE=${{ steps.get_latest_release.outputs.release }}
          TAG=${{ steps.get_latest_release.outputs.tag }}
          if [ -z ${{ steps.get_latest_release.outputs.version }} ]; then 
            VERSION=${TAG#v}
          else 
            VERSION=${{ steps.get_latest_release.outputs.version }}
          fi
        else
          RELEASE=1
          TAG=${{ github.event.release.tag_name }}
          VERSION=${TAG#v}
        fi
        echo "VERSION=$VERSION" >> $GITHUB_OUTPUT
        echo "RELEASE=$RELEASE" >> $GITHUB_OUTPUT
        echo "TAG=$TAG" >> $GITHUB_OUTPUT
```